tests/Unit/inc/classes/ImagifyUser/TestIsOverQuota.php

```diff
@@ -0,0 +1,140 @@
+			'plan_label'                   => 'whatever',
+			'quota'                        => 456,
+			'extra_quota'                  => 0,
+			'extra_quota_consumed'         => 0,
+			'consumed_current_month_quota' => 123,
+			'next_date_update'             => '',
+			'is_active'                    => 1,
+		];
+
+		Functions\when( 'get_imagify_user' )->justReturn( $userData );
+
+		$this->assertFalse( ( new Imagify_User() )->is_over_quota() );
+	}
+
+	/**
+	 * Test Imagify_User->is_over_quota() should return false when free and not over quota.
+	 */
+	public function testShouldReturnFalseWhenFreeNotOverQuota() {
+		$userData = (object) [
+			'id'                           => 1,
+			'email'                        => 'imagify@example.com',
+			'plan_id'                      => '1',
+			'plan_label'                   => 'free',
+			'quota'                        => 1000,
+			'extra_quota'                  => 0,
+			'extra_quota_consumed'         => 0,
+			'consumed_current_month_quota' => 900, // Current consumed quota 90%.
+			'next_date_update'             => '',
+			'is_active'                    => 1,
+		];
+
+		$this->createMocks( $userData, 90 );
+
+		$this->assertFalse( ( new Imagify_User() )->is_over_quota() );
+	}
+
+	/**
+	 * Test Imagify_User->is_over_quota() should return true when free and over quota.
+	 */
+	public function testShouldReturnTrueWhenFreeOverQuota() {
+		$userData = (object) [
+			'id'                           => 1,
+			'email'                        => 'imagify@example.com',
+			'plan_id'                      => '1',
+			'plan_label'                   => 'free',
+			'quota'                        => 1000,
+			'extra_quota'                  => 0,
+			'extra_quota_consumed'         => 0,
+			'consumed_current_month_quota' => 1000, // Current consumed quota 100%.
+			'next_date_update'             => '',
+			'is_active'                    => 1,
+		];
+
+		$this->createMocks( $userData, 100 );
+
+		$this->assertTrue( ( new Imagify_User() )->is_over_quota() );
+	}
+
+	private function createMocks( $userData, $dataPreviousQuotaPercent ) {
+		Functions\when( 'get_imagify_user' )->justReturn( $userData );
+		Functions\when( 'imagify_round_half_five' )->alias(
+			function( $number ) {
+				$number = strval( $number );
+				$number = explode( '.', $number );
+
+				if ( ! isset( $number[1] ) ) {
+					return $number[0];
+				}
+
+				$decimal = floatval( '0.' . substr( $number[1], 0, 2 ) ); // Cut only 2 numbers.
+
+				if ( $decimal > 0 ) {
+					if ( $decimal <= 0.5 ) {
+						return floatval( $number[0] ) + 0.5;
+					}
+					if ( $decimal <= 0.99 ) {
+						return floatval( $number[0] ) + 1;
+					}
+					return 1;
+				}
+
+				return floatval( $number );
+			}
+		);
+
+		$imagify_data_mock = Mockery::mock( Imagify_Data::class );
+		// Change the Imagify_Data::$_instance to the mock.
+		$ref = $this->get_reflective_property( '_instance', Imagify_Data::class );
+
+		$ref->setValue( $imagify_data_mock, $imagify_data_mock );
```

Comment:
This code could be moved to the trait for reuse throughout the tests.

```php
$this->set_singleton_instance( Imagify_Data::class, $imagify_data_mock );
```

And the method on the Trait:

```php

/**
 * Set the singleton's `$_instance` property to the given instance.
 *
 * @param string $class_name Name of the target class.
 * @param mixed  $instance   Instance of the target object.
 *
 * @throws \ReflectionException Throws an exception if property does not exist.
 */
protected function set_singleton_instance( $class, $instance ) {
	$ref = $this->get_reflective_property( '_instance', $class );
	$ref->setValue( $instance, $instance );
}
```